mousestyles/dynamics/tests/test_dynamics.py

```diff
@@ -37,3 +41,59 @@ def test_creat_time_matrix():
     # Checking functions output the correct time matrix
     matrix = create_time_matrix(combined_gap=4, time_gap=1, days_index=0)
     assert matrix.iloc[0, 2181] == 1.0
+
+
+def test_get_prob_matrix_list_input():
+    # checking functions raise the correct errors for wrong input
+    # time_df is not DataFrame
+    with pytest.raises(ValueError) as excinfo:
+        get_prob_matrix_list(time_df=5, interval_length=1000)
+    assert excinfo.value.args[0] == "time_df should be pandas DataFrame"
+    # interval_length is 0
+    row_i = np.hstack((np.zeros(13), np.ones(10),
+                      np.ones(10)*2, np.ones(10)*3))
+    time_df_eg = np.vstack((row_i, row_i, row_i))
+    time_df_eg = pd.DataFrame(time_df_eg)
+    with pytest.raises(ValueError) as excinfo:
+        get_prob_matrix_list(time_df=time_df_eg, interval_length=0)
+    assert excinfo.value.args[0] == "combined_gap should be positive int"
+    # interval_length is not int
+    with pytest.raises(ValueError) as excinfo:
+        get_prob_matrix_list(time_df=time_df_eg, interval_length=0.5)
+    assert excinfo.value.args[0] == "combined_gap should be positive int"
+
+
+def test_get_prob_matrix_list():
+    # Checking functions output the correct matrix list
+    row_i = np.hstack((np.zeros(13), np.ones(10),
+                      np.ones(10)*2, np.ones(10)*3))
+    time_df_eg = np.vstack((row_i, row_i, row_i))
+    time_df_eg = pd.DataFrame(time_df_eg)
+    mat_list = get_prob_matrix_list(time_df_eg,
+                                    interval_length=10)
+    assert mat_list[0][0, 0] == 1.
+    assert sum(sum(mat_list[0])) == 1.
+
+
+def test_get_prob_matrix_small_interval_input():
+    # checking functions raise the correct errors for wrong input
+    # string_list is not list
+    with pytest.raises(ValueError) as excinfo:
+        get_prob_matrix_small_interval(string_list=np.array([1, 2]))
+    assert excinfo.value.args[0] == "string_list should be a list"
+    # items in string_list is not string
+    time_list = [0, 1, 2]
+    with pytest.raises(ValueError) as excinfo:
+        get_prob_matrix_small_interval(string_list=time_list)
+    assert excinfo.value.args[0] == "items in string_list should be str"
+
+
+def test_get_prob_matrix_small_interval():
+    # Checking functions output the correct matrix
+    time_list = ['002', '001', '012']

```

Comment:
time_list is a simple example where the object is a list but each element is not string.